src/main.rs

```diff
@@ -242,6 +266,11 @@ fn main_send(addr: &str, fnames: &[String]) -> Result<()> {
     let listener = std::net::TcpListener::bind(addr)?;
 
     println!("Waiting for the receiver ...");
+    sender_events
+        .send(SenderEvents::Listening(
+            listener.local_addr().unwrap().port(),
+        ))
+        .map_err(|x| Error::new(ErrorKind::Other, x))?;
```

Comment:
```suggestion
        .expect("Listener should not exit before the sender.");
```

Sending fails if the receiving end of the channel has been dropped. That is a programming error, not an expected runtime error, so I would make it crash rather than propagating an error that we have no good way of handling.